openssl/src/x509/mod.rs

```diff
@@ -1028,6 +1028,21 @@ impl X509NameRef {
         }
     }
 
+    /// Compare two names, like [`Ord`] but it may fail.
+    ///
+    /// With OpenSSL versions from 3.0.0 this may return an error if the underlying `X509_NAME_cmp`
+    /// call fails.
+    /// For OpenSSL versions before 3.0.0 it will never return an error, but due to a bug it may
+    /// spuriously return `Ordering::Less` if the `X509_NAME_cmp` call fails.
+    #[corresponds(X509_NAME_cmp)]
+    pub fn try_cmp(&self, other: &X509NameRef) -> Result<Ordering, ErrorStack> {
+        let cmp = unsafe { ffi::X509_NAME_cmp(self.as_ptr(), other.as_ptr()) };
+        if cfg!(ossl300) && cmp == -2 {
```

Comment:
It looks like X509_NAME_cmp is documented to return -2 on failure as well in 1.1.1 at least: https://www.openssl.org/docs/man1.1.1/man3/X509_NAME_cmp.html

Reply:
It does return -2 on failure, however [this line](https://github.com/openssl/openssl/blob/OpenSSL_1_1_1-stable/crypto/x509/x509_cmp.c#L183) means that it also returns -2 if the first argument is two bytes shorter than the second. That issue is what I was trying to get at in the doc comment, if that's not clear do you have any suggestions to improve it?

Reply:
Wow. I guess that works then...